src/Common/Logger.cpp

```diff
@@ -1,29 +1,112 @@
-#include <Poco/Logger.h>
 #include <Common/Logger.h>
+#include <Common/Exception.h>
+#include <Loggers/OwnPatternFormatter.h>
+#include <Loggers/TextLogSink.h>
 
-LoggerPtr getLogger(const std::string & name)
+#include <quill/Frontend.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+}
+}
+namespace
+{
+std::unique_ptr<OwnPatternFormatter> formatter;
+std::atomic<OwnPatternFormatter *> formatter_ptr = nullptr;
+}
+
+Logger::Logger(std::string_view name_, QuillLoggerPtr logger_)
+    : name(name_)
+    , logger(logger_)
+{
+}
+
+Logger::Logger(std::string name_, QuillLoggerPtr logger_)
+{
+    name_holder = std::move(name_);
+    name = name_holder;
+    logger = logger_;
+}
+
+OwnPatternFormatter * Logger::getFormatter()
+{
+    return formatter_ptr.load(std::memory_order_relaxed);
+}
+
+void Logger::setFormatter(std::unique_ptr<OwnPatternFormatter> formatter_)
+{
+    formatter = std::move(formatter_);
+    formatter_ptr.store(formatter.get(), std::memory_order_relaxed);
+}
+
+QuillLoggerPtr Logger::getQuillLogger()
+{
+    if (!logger)
+        logger = quill::Frontend::get_logger("root");
+
+    return logger;
+}
+
+DB::TextLogSink & Logger::getTextLogSink()
 {
-    return Poco::Logger::getShared(name);
+    static DB::TextLogSink text_log_sink;
+    return text_log_sink;
 }
 
-LoggerPtr createLogger(const std::string & name, Poco::Channel * channel, Poco::Message::Priority level)
+LoggerPtr getLogger(const char * name, const char * component_name)
 {
-    return Poco::Logger::createShared(name, channel, level);
+    return getLogger(std::string_view{name}, component_name);
 }
 
-LoggerRawPtr getRawLogger(const std::string & name)
+LoggerPtr getLogger(std::string_view name, const char * component_name)
 {
-    return &Poco::Logger::get(name);
+    auto * logger = quill::Frontend::get_logger("root");
```

Comment:
In the logging library, `get_logger()` is a thread-safe function that acquires a mutex lock and performs a hash map lookup. This might be acceptable for your case. However, it's worth noting that if you want to avoid this overhead, consider caching the raw pointer to the logger. The `Logger*` will remain stable even if new loggers are added, eliminating the need to look it up multiple times.